designs/node-overlay-dra.md

```diff
@@ -0,0 +1,564 @@
+# NodeOverlay Integration for Dynamic Resource Allocation
+
+## Overview
+
+When Karpenter constructs a NodeClaim for a given set of pods, it uses information from the cloudprovider to determine
+the resources that will exist on a Node if one is created for a given instance type. That is then used to determine
+which instance types will satisfy a given set of requests. Cloudprovider implementations can get this information from
+any number of sources, for example AWS EC2's `DescribeInstanceTypes` API, and they use their knowledge of well-known
+device plugins to determine the resulting nodes' allocatable resources. The drawback of this approach is that each
+cloudprovider needs to build integrations for specific drivers, and users are limited by those integrations.
```

Comment:
While I agree with this statement, there's an argument to be made that the rising adoption of GPUs (and their corresponding drivers) need to be treated as first-class citizens rather than advanced features.